frontend/public/components/utils/dropdown.jsx

```diff
@@ -285,7 +285,18 @@ class Dropdown_ extends DropdownMixin {
   applyTextFilter_(autocompleteText, items) {
     const { autocompleteFilter } = this.props;
     if (autocompleteFilter && !_.isEmpty(autocompleteText)) {
-      items = _.pickBy(items, (item, key) => autocompleteFilter(autocompleteText, item, key));
+      const newItems = [];
+      let totalItems = _.map(items, (item) => item.props);
+      totalItems = autocompleteFilter(autocompleteText, totalItems);
```

Comment:
Will this break `autocompleteFilter` with other non-list dropdowns? We've changed the signature. It appears to be used in a lot of places.

Reply:
I would call this something like `result` instead of `totalItems`

Reply:
> Will this break `autocompleteFilter` with other non-list dropdowns? We've changed the signature. It appears to be used in a lot of places.

On checking again, it seems like it will.
I was thinking that probably we can use another prop like `weightedSort={true}` and check the same in the dropdown code and usefuzzysort/search accordingly. That would give us the flexibility to use both of them as per need. What do you think?

Reply:
If the filter functions have different signature, I'd make them different props. Maybe `autocompleteFilter` and `weightedSortFilter`?

Reply:
Yes, I meant that only.
Updated the code with the same.